### PR TITLE
Feature/rpgmd 12 enemy combat

### PR DESCRIPTION
**Pull request by Dreshpack**
vse gotovo

### CODE REVIEW

File: Assets/Scripts/HeroScripts/CombatScript.cs [hunk 1]
```diff
@@ -7,10 +7,21 @@ public class CombatScript : MonoBehaviour
...
-    private UnityEvent<bool> attackPressed;
+    private MovingScript playerMovement;
+    private Rigidbody2D rigidBody;
+    private Enemy AttackingEnemy;
     private bool attacks = false;
+    private float lastInputTime;
+    private bool damageable = true;
+    private bool isDead = false;
+
+
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/CombatScript.cs]: Delete this empty line

File: Assets/Scripts/Enemy/Enemy.cs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    private int currentHealth;
+    private bool isTakingDamage;
+    private bool isDead;
+    public GameObject GetPlayer
+    {
+        get
+        {
+            return player[0];
+        }
+    }
```
Comment (Hlib-mats) [inline Assets/Scripts/Enemy/Enemy.cs:39]: Follow conventions. Add lines between methods 
Refactor ur code(just naming and smth like adding or deleting empty lines)

File: Assets/Scripts/HeroScripts/CombatScript.cs [hunk 3]
```diff
@@ -20,6 +31,7 @@ public void attack()
             _animationController.SetTrigger("attack");
             _weaponCollider.enabled = !_weaponCollider.enabled;
             StartCoroutine(AttackSpeedController());
+
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/CombatScript.cs:31]: Delete this line

File: Assets/Scripts/HeroScripts/CombatScript.cs [hunk 4]
```diff
@@ -38,12 +50,62 @@ private int dealingDamage()
...
+            {
+                rigidBody.AddForce(Vector2.right * horizontalKnockback);
+            }
+            if (!isDead)
+            {
+                _animationController.SetTrigger("hit");
+                Invoke("EnableMovement", cancelMovementTime);
+                Invoke("CancelHit", invulnerabilityTime);
+            }
+
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/CombatScript.cs]: Also delete

File: Assets/Scripts/HeroScripts/CombatScript.cs [hunk 5]
```diff
@@ -38,12 +50,62 @@ private int dealingDamage()
...
+    private void CancelHit()
+    {
+        attacks = false;
+    }
+    public void EnableMovement()
+    {
+        playerMovement.IsTakingDamage = false;
+    }
     private IEnumerator AttackSpeedController()
+
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/CombatScript.cs:80]: Wth is this ?

File: Assets/Scripts/HeroScripts/CombatScript.cs [hunk 6]
```diff
@@ -55,5 +117,10 @@ private void Start()
     {
         _player = GetComponentInParent<Player>();
         _weaponCollider = GetComponentInParent<CapsuleCollider2D>();
+        rigidBody = GetComponentInParent<Rigidbody2D>();
+        playerMovement = GetComponentInParent<MovingScript>();
+    }
+    private void FixedUpdate()
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/CombatScript.cs]: Delete this method, if there is no references

File: Assets/Scripts/HeroScripts/MovingScript.cs [hunk 7]
```diff
@@ -33,14 +34,35 @@ public DirectionState GetDirectionState
...
+    {
+        get
+        {
+            return isTakingDamage;
+        }
+        set
+        {
+            isTakingDamage = value;
+        }
+    }
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/MovingScript.cs:47]: Also refactoring this code by conventions

File: Assets/Scripts/HeroScripts/MovingScript.cs [hunk 8]
```diff
@@ -105,6 +127,9 @@ private void Reflect()
             }
         }
     }
+    private void FixedUpdate()
```
Comment (Hlib-mats) [inline Assets/Scripts/HeroScripts/MovingScript.cs]: Also please delete if there is no references

**Review by Hlib-mats — COMMENTED** [review]
Refactor ur code by conventions and delete unnecessary methods

File: Assets/Scripts/HeroScripts/CombatScript.cs (hunk 5 above)
Comment (Dreshpack) [reply Assets/Scripts/HeroScripts/CombatScript.cs:80]: function for coroutine

**Review by Hlib-mats — APPROVED** [review]
approving for now, need to be refactored

**Pull request merged**